spark_client/cli/spark-shell.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import re
+from typing import Optional
+
+from spark_client.cli import defaults
+from spark_client.domain import ServiceAccount
+from spark_client.services import (
+    K8sServiceAccountRegistry,
+    KubeInterface,
+    SparkInterface,
+)
+
+if __name__ == "__main__":
```

Comment:
Not a big deal, but the setup code looks duplicated between `spark-shell.py` and `spark-submit.py`. Would've been good to respect the DRY principal if at all possible; however I'm not sure if there's a way to extract this stuff into a shared method, perhaps there isn't in Python, idk.

Reply:
Uhm. It is a good point. I believe there should be a way and I have done this in the past. @averma-canonical I will give you some references for this